src/indieweb_utils.py

```diff
@@ -42,21 +51,16 @@ def canonicalize_url(url, domain, full_url=None, protocol="https"):
     """
 
     if url.startswith("http://") or url.startswith("https://"):
-        domain = url.split("/")[2]
-
-        # remove port from domain
-
-        domain = domain.split(":")[0]
-        protocol = url.split("/")[0]
-
-        return protocol + "//" + domain + "/" + "/".join(url.split("/")[3:])
-
-    if ":" in domain:
-        text_before_port = domain.split(":")[0]
+        domain = url_parse.urlsplit(url).netloc
+        url_protocol = url_parse.urlsplit(url).scheme
+        path = url_parse.urlsplit(url).path
 
-        text_after_port = domain.split(":")[1].split("/")[0]
+        if url.startswith("http://"):
+            domain = domain.replace(":80", "")
+        elif url.startswith("https://"):
+            domain = domain.replace(":443", "")
```

Comment:
We probably don't want to hard code the ports here as you could be running http server on any port. You can just split by `":"` like you had before. No need to add the if statement to check for : as the domain will always be the first object.

```python
>>> url_parse.urlsplit("http://jamesvandyne.com:80/now").netloc.split(":")[0]
'jamesvandyne.com'
>>> url_parse.urlsplit("http://jamesvandyne.com/now").netloc.split(":")[0]
'jamesvandyne.com'
```